File: 2do_DAM/PMDM/1ra_Evaluacion/Ejercicios/Java/Ejercicio08_Baloncesto/app/src/main/java/com/example/ejer8_baloncesto/MainActivity.java
package com.example.ejer8_baloncesto;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Button;
import android.widget.LinearLayout;

import com.example.ejer8_baloncesto.databinding.ActivityMainBinding;
import com.google.android.material.button.MaterialButton;

public class MainActivity extends AppCompatActivity {

    ActivityMainBinding binding;
    int puntosvisit=0;
    int puntoslocal=0;
    int jugadores=14;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        //setContentView(R.layout.activity_main);
        binding=ActivityMainBinding.inflate(getLayoutInflater());
        setContentView(binding.getRoot());
        marcador();
        inicializar(binding.lyLocales);
        inicializar(binding.lyVisit);
    }

    private void marcador(){
        binding.tvMarcador.setText(getResources().getString(R.string.tantero,puntoslocal,puntosvisit));
    }


    public void inicializar(LinearLayout contenedor)
    {
        LinearLayout.LayoutParams parametros= new LinearLayout.LayoutParams(150, 150);
        for (int i=4;i<=jugadores;i++){
            Button boton=new Button(this);
            boton.setText(""+i);
            boton.setLayoutParams(parametros);
            contenedor.addView(boton);
            boton.setOnClickListener(new View.OnClickListener() {
                @Override
                public void onClick(View v) {
                    if (binding.rbLibre.isChecked()){
                        if (((LinearLayout)v.getParent()).getId()==binding.lyLocales.getId())
                                puntoslocal+=1;
                        else
                                puntosvisit+=1;
                    }else if(binding.rbCanasta.isChecked()){
                        if (((LinearLayout)v.getParent()).getId()==binding.lyLocales.getId())
                            puntoslocal+=2;
                        else
                            puntosvisit+=2;
                    }else if(binding.rbTriple.isChecked()){
                        if (((LinearLayout)v.getParent()).getId()==binding.lyLocales.getId())
                            puntoslocal+=3;
                        else
                            puntosvisit+=3;
                    }
                    marcador();
                }
            });
        }
    }

}

File: 2do_DAM/PMDM/1ra_Evaluacion/Ejemplos/Kotlin/Ejemplo015_Alertas/app/src/main/java/com/example/ejem15_alertas/MainActivity.kt
package com.example.ejem15_alertas

import android.R
import android.os.Bundle
import android.util.Log
import android.view.View
import android.widget.ImageView
import androidx.appcompat.app.AlertDialog
import androidx.appcompat.app.AppCompatActivity
import com.bumptech.glide.Glide
import com.example.ejem15_alertas.databinding.ActivityMainBinding
import com.example.ejem15_alertas.databinding.CustomDialogBinding
import com.google.android.material.snackbar.Snackbar


class MainActivity : AppCompatActivity() {

    lateinit var binding : ActivityMainBinding

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        //setContentView(R.layout.activity_main)
        binding = ActivityMainBinding.inflate(layoutInflater)
        setContentView(binding.root)

        binding.bAlerta.setOnClickListener( View.OnClickListener {
            val builder = AlertDialog.Builder(this)
            builder.setTitle("Ventana alerta")
            builder.setMessage("Ejemplo de alerta")
            builder.setPositiveButton("OK") { dialog, which ->
                Snackbar.make(binding.root, "Has pulsado OK", Snackbar.LENGTH_LONG).show()
                dialog.cancel()
            }
            builder.setNegativeButton("Cancelar"){ dialog, which ->
                Snackbar.make(binding.root, "Has pulsado Cancelar", Snackbar.LENGTH_LONG).show()
                dialog.cancel()
            }
            builder.show()
        })

        binding.bSeleccionMultiple.setOnClickListener( View.OnClickListener {
            val listaOpciones: Array<String> = arrayOf("Opcion 1", "Opcion 2", "Opcion 3")
            val builder = AlertDialog.Builder(this)
         builder.setTitle("Ventana alerta")
            builder.setItems(listaOpciones) { dialog, which ->
                Snackbar.make(
                    binding.root,
                    "Has pulsado la opcion " + listaOpciones[which],
                    Snackbar.LENGTH_LONG
                ).show()
                dialog.cancel()
            }
         builder.show()

        })

        Glide.with(this).load("https://picsum.photos/400/400").into(binding.ivImagen)

        binding.bDialogoPersonalizado.setOnClickListener( View.OnClickListener {

            val builder = AlertDialog.Builder(this)
            val inflater =  layoutInflater
            var binding_customDialog = CustomDialogBinding.inflate(inflater)
            var dialogView=binding_customDialog.root

            Glide.with(this).load("https://picsum.photos/80/180").into(binding_customDialog.ivImagenAlertPersonalizado)

            builder.setView(dialogView)
            builder.setTitle("Diágolo personalizado de Alerta")
            builder.setPositiveButton("OK") { dialog, which ->
                var nombre : String =binding_customDialog.titNombre.text.toString()
                var soltero : Boolean = binding_customDialog.cbEstadoCivil.isChecked
                var ciudad : String = binding_customDialog.sCiudadNacimiento.selectedItem.toString()

                var usuario = Usuario(nombre, soltero, ciudad)
                Log.d("depurando", usuario.toString())
                binding.usuario = usuario

                Snackbar.make(binding.root, "Has pulsado OK", Snackbar.LENGTH_LONG).show()
                dialog.cancel()
            }

            builder.show()

        })



    }
}